### DISCUSSION TITLE
If you're seeing this, something is embarrassingly wrong

### DISCUSSION THREAD

**Discussion by jeremyjyang** [nih-cfde/published-documentation#334] [Q&A] [answered]
Category: FYI/WTF. 

Got this error message yesterday (3/15). "Beware the ides of March." comes to mind...

>  Loading all CV terms used in submission tables... [Tue 15 Mar 2022 09:37:48 AM MDT]
> 
>    scanning "/home/data/CFDE/data/targetpages6124/submission/file.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/biosample.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/biosample_disease.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/biosample_gene.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/biosample_substance.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/subject_disease.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/subject_role_taxonomy.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/subject_substance.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/collection_disease.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/phenotype_disease.tsv"...
>    scanning "/home/data/CFDE/data/targetpages6124/submission/phenotype_gene.tsv"...
> 
> ...done scanning all CV terms used in this submission. [Tue 15 Mar 2022 09:37:48 AM MDT]
> Loading display data (names, descriptions, etc.) for all CV terms used in submission tables... [Tue 15 Mar 2022 09:37:49 AM MDT]
> 
>    Uber-anatomy ontology... [Tue 15 Mar 2022 09:37:49 AM MDT]
>    Ontology for Biomedical Investigations (assay type)... [Tue 15 Mar 2022 09:37:50 AM MDT]
>    Ontology for Biomedical Investigations (analysis type)... [Tue 15 Mar 2022 09:37:50 AM MDT]
>    EDAM (data type)... [Tue 15 Mar 2022 09:37:50 AM MDT]
>    Disease Ontology... [Tue 15 Mar 2022 09:37:50 AM MDT]
>    EDAM (file format)... [Tue 15 Mar 2022 09:37:51 AM MDT]
>    NCBI Taxonomy... [Tue 15 Mar 2022 09:37:51 AM MDT]
>    Ensembl genes... [Tue 15 Mar 2022 09:37:57 AM MDT]
>    PubChem substances and compounds [if you included PubChem terms in your submission, this step will take around 10 minutes, even with a fast disk]...
>       ...processing substance file... [Tue 15 Mar 2022 09:37:57 AM MDT]
>       ...processing compound file... [Tue 15 Mar 2022 09:37:58 AM MDT]
> 
> ...done loading display data for all CVs. [Tue 15 Mar 2022 09:37:58 AM MDT]
> Writing CV term tracker tables (to be included in complete C2M2 submission)... [Tue 15 Mar 2022 09:37:58 AM MDT]
> 
>    writing "/home/data/CFDE/data/targetpages6124/submission/anatomy.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/assay_type.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/analysis_type.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/data_type.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/disease.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/file_format.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/ncbi_taxonomy.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/gene.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/substance.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
>    writing "/home/data/CFDE/data/targetpages6124/submission/compound.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
> 
> ...done writing term tables for final C2M2 submission datapackage. [Tue 15 Mar 2022 09:37:58 AM MDT]
> NOTE: HPO phenotype data processing is currently offline; stubs will be generated for basic format compliance, but non-null phenotype code values will be ignored and may cause submission errors.
> Writing "/home/data/CFDE/data/targetpages6124/submission/phenotype.tsv"... [Tue 15 Mar 2022 09:37:58 AM MDT]
> Ensuring all file.tsv records with persistent IDs have non-null checksums... [Tue 15 Mar 2022 09:37:58 AM MDT]
> 
> ...done; all file records passed checksum verification. [Tue 15 Mar 2022 09:37:59 AM MDT]
> Ensuring all persistent IDs are unique (both within and across tables)... [Tue 15 Mar 2022 09:37:59 AM MDT]
> 
>    FATAL: If you're seeing this, something is embarrassingly wrong

**Comment by abradyIGS** [comment, marked as answer]
Changed error message to "If you're seeing this, you're missing a required `persistent_id` column in one or more of file.tsv, biosample.tsv, subject.tsv, collection.tsv or project.tsv," which is what it should've said all along -- sorry for crypticity.

Please check for required persistent_id column in all listed tables (values can be null, but column must be present); if it's there in all of them, please send copies of all five tables, as parsing will have failed in an unknown manner
